### PR TITLE
merge Engine with Server

### PR DESCRIPTION
**Pull request by snowAvocado**
# Ticket(s)

## Development Checklist

- [x] I have added a descriptive title to this PR.
- [x] I have squashed related commits together.
- [x] I have rebased my branch on top of the latest main branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added docstring(s) and type annotations to my code.
- [x] I have made corresponding changes to the documentation (docs).
- [x] I have added tests for my changes.

## Legal Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/gatewayd-io/gatewayd/blob/main/CONTRIBUTING.md) document.
- [x] I have read and understood the [Code of Conduct](https://github.com/gatewayd-io/gatewayd/blob/main/CODE_OF_CONDUCT.md).
- [x] I have read and agreed to the [Apache CLA](https://www.apache.org/licenses/contributor-agreements.html) (required).
- [x] I have read and agreed to the [LICENSE](https://github.com/gatewayd-io/gatewayd/blob/main/LICENSE) (required).

### CODE REVIEW

**Comment by snowAvocado** [comment]
#347

File: network/server.go [hunk 1]
```diff
@@ -650,3 +656,33 @@ func NewServer(
...
+
+// CountConnections returns the current number of connections.
+func (s *Server) CountConnections() int {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return int(s.connections)
+}
+
+// Stop stops the server.
+func (s *Server) Stop() error {
```
Comment (mostafa) [inline network/server.go:668]: Unexport this function (make it internal) and remove it from the interface, as we don't want to stop the server without stopping the attached proxy instance, as the result would be unknown behavior. Optionally you can merge it with the `Shutdown` method. WDYT?

**Review by mostafa — COMMENTED** [review]
Hey @snowAvocado,

Thanks for your contribution! 🙏 I have two comments:
1. Please fix linter errors (and ignore unimportant ones).
2. You can merge the stop with shutdown and unexport the stop method, effectively removing it from the interface.

File: network/server.go (hunk 1 above)
Comment (snowAvocado) [reply network/server.go]: @mostafa Thanks for your swift review, i pushed new commit with your comments

**Review by mostafa — APPROVED** [review]
LGTM! :rocket:

**Comment by mostafa** [comment]
@snowAvocado Please sign your commits.

**Comment by snowAvocado** [comment]
Hi @mostafa , I think signing commits changes hash-ids of my commits  and diverge my local from this remote branch . since your commit is top on my commits, would i be able to push your change as you are the author of commit ? for now i have created new pull request my two commits, i could not manage to find a proper work around this.

**Review by mostafa — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**